### PR TITLE
first round of committing

### PR DESCRIPTION
**Pull request by FeiYeYe**
@robertzk is the usage pattern in the tests reasonable for users?

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+    # dbWriteTable(dbconn, tblname, df, row.names = 0, append = TRUE)
+    # Believe it or not, the above does not work! RPostgreSQL seems to have a
+    # bug that incorrectly serializes some kinds of data into the database.
+    # Thus we must roll up our sleeves and write our own INSERT query. :-(
+    number_of_records_per_insert_query <- 250
+    slices <- slice(seq_len(nrow(df)), nrow(df) / number_of_records_per_insert_query)
+
+    # If we don't do this, we will get really weird bugs with numeric things stored as character
+    # For example, a loan with ID 100000 will be stored as 10e+5, which is wrong.
```
Comment (peterhurford) [inline R/utils.R]: Might not want to talk about loans in an open source package.

**Comment by peterhurford** [comment]
You can get inspiration for more tests here: https://github.com/peterhurford/batchman/blob/master/inst/tests/test-batch.R

File: R/utils.R (hunk 1 above)
Comment (FeiYeYe) [inline R/utils.R]: lol, this is copied over from avant

**Comment by FeiYeYe** [comment]
@robertzk looking better? I think it does what we assume by name it would do now.

File: R/db.R [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+#' @param df data frame. The data frame to insert.
+#' @importFrom DBI dbWriteTable
+#' @importFrom DBI dbReadTable
+#' @importFrom DBI dbRemoveTable
+dbWriteTableUntilSuccess <- function(dbconn, tblname, df) {
+  dbRemoveTable(dbconn, tblname)
+  success <- FALSE
+  while (!success) {
+    dbWriteTable(dbconn, tblname, df, append = FALSE, row.names = 0)
+    df_db <- dbReadTable(dbconn, tblname)
```
Comment (robertzk) [inline R/db.R]: This is going to be insanely slow on large data sets. Need to bring back the integrity checks with warnings.

**Comment by FeiYeYe** [comment]
nvm

**Pull request merged**